app/models/topology.rb

```diff
@@ -92,4 +93,14 @@ def validate_stakeholders
       end
     end
   end
+
+  def validate_clone
+    return unless original.present?
+    return unless each_node
+
+    if each_node.count != original.each_node.count
```

Comment:
I just realised `Enumerator` doesn't implement `#length`, so that's why you used `#count`. Sorry, my bad.